wallet/import.go

```diff
@@ -1,13 +1,191 @@
 package wallet
 
 import (
+	"encoding/binary"
+	"errors"
 	"fmt"
 
 	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcutil/hdkeychain"
 	"github.com/btcsuite/btcwallet/waddrmgr"
 	"github.com/btcsuite/btcwallet/walletdb"
 )
 
+// pubKeyHDVersion defines the different public key HD versions supported that
+// can be imported into the wallet.
+type pubKeyHDVersion uint32
+
+const (
+	mainnetBIP0044 pubKeyHDVersion = 0x0488b21e // xpub
+	testnetBIP0044 pubKeyHDVersion = 0x043587cf // tpub
+
+	mainnetBIP0049Plus pubKeyHDVersion = 0x049d7cb2 // ypub
```

Comment:
Note that bip49 plus was somethign i actually made up. If someone imports an actual "ypub" that used nested p2sh for its change addresses, then we'd miss the funds on rescan.

Reply:
TIL. This seems like a problem then, how can we differentiate the two? We could leave it up to the user to explicitly tell us if it's originating from another wallet, in which case we'd use and scan for P2SH change addresses, but is there anything better we could do?

Reply:
Yeah back then, it didn't make sense to me to still use a nested p2wkh address there, given it was a _change_ address. We'd either not support "ypub" (still makes me lol tbh), or also add in "proper" BIP 49, which is just a new key scope definition:
```go
KeyScopeBIP0049 = KeyScope{
    Purpose: 49, 
    Coin: 0, 
}

ScopeAddrMap = map[KeyScope]ScopeAddrSchema{
    KeyScopeBip0049: {
        ExternalAddrType: NestedWitnessPubKey, 
        InternalAddrType: NestedWitnessPubKey,
    }, 
}
```

Reply:
Resolved this offline as we'll just modify the existing scope since wallet instances today don't ever use that scope.